mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -226,55 +225,95 @@ impl State for WaitForTakerPaymentSpend {
     type Result = ();
 
     async fn on_changed(self: Box<Self>, watcher_ctx: &mut WatcherContext) -> StateResult<Self::Ctx, Self::Result> {
-        if std::env::var("SWAP_WATCHER_SKIP_WAITING").is_err() {
-            // Sleep for half the locktime to allow the taker to spend the maker payment first
-            let now = now_ms() / 1000;
-            let wait_for_taker_until =
-                wait_for_taker_payment_conf_until(watcher_ctx.data.swap_started_at, watcher_ctx.data.lock_duration);
-            let sleep_duration = (wait_for_taker_until - now + 1) as f64;
-
-            if now < wait_for_taker_until {
-                Timer::sleep(sleep_duration).await;
-            }
-        }
-
-        let f = watcher_ctx.taker_coin.wait_for_htlc_tx_spend(
-            &watcher_ctx.data.taker_payment_hex,
-            &[],
-            watcher_ctx.data.taker_payment_lock,
-            watcher_ctx.data.taker_coin_start_block,
-            &None,
-        );
-
-        let tx = match f.compat().await {
-            Ok(t) => t,
-            Err(err) => {
-                error!("{}", err.get_plain_text_format());
-                return Self::change_state(RefundTakerPayment {});
-            },
+        const TAKER_PAYMENT_SPEND_SEARCH_INTERVAL_SLOW: f64 = 300.;
+        let payment_search_interval = match std::env::var("USE_TEST_SEARCH_INTERVAL") {
+            Ok(_) => TAKER_PAYMENT_SPEND_SEARCH_INTERVAL,
+            Err(_) => TAKER_PAYMENT_SPEND_SEARCH_INTERVAL_SLOW,
         };
-
-        let tx_hash = tx.tx_hash();
-        info!("Taker payment spend tx {:02x}", tx_hash);
-        let tx_ident = TransactionIdentifier {
-            tx_hex: BytesJson::from(tx.tx_hex()),
-            tx_hash,
+        let wait_until = match std::env::var("SKIP_WAIT_FOR_TAKER_PAYMENT") {
+            Ok(_) => 0,
+            Err(_) => watcher_ctx.data.swap_started_at + (4 * watcher_ctx.data.lock_duration / 3),
+        };
+        let search_input = WatcherSearchForSwapTxSpendInput {
+            time_lock: (watcher_ctx.data.swap_started_at + watcher_ctx.data.lock_duration) as u32,
```

Comment:
`watcher_ctx.data.swap_started_at + watcher_ctx.data.lock_duration` is duplicated several times. Could you wrap it in function, please?